### DISCUSSION TITLE
Where do Bandcamp recommendations come from?

### DISCUSSION THREAD

**Discussion by mivuzu** [TeamNewPipe/NewPipe#10540] [Questions] [answered]
I'm just curious given to the obvious fact bandcamp doesn't have any recommendations and the ones on this app are noticeably good

**Comment by AudricV** [comment, marked as answer]
> given to the obvious fact bandcamp doesn't have any recommendations

The service has recommendations, look at the bottom of a track page on the website. That's [where recommendations are extracted by NewPipe's extractor](https://github.com/TeamNewPipe/NewPipeExtractor/blob/eac850ca1077c7615d8b211502f346eb5e2284b0/extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampStreamExtractor.java#L188-L197).
